lib/declare_schema/model.rb

```diff
@@ -192,7 +192,18 @@ def add_validations_for_field(name, type, args, options)
         end
       end
 
-      def add_formatting_for_field(name, type, _args)
+      def add_serialize_for_field(name, type, options)
+        if (serialize_value = options.delete(:serialize))
+          type == :string || type == :text or raise ArgumentError, "serialize type must be :string or :text"
+          serialize_args = Array((serialize_value unless serialize_value == true))
+          serialize(name, *serialize_args)
+          if options.has_key?(:default) && !options[:default].is_a?(String) && !options[:default].nil?
+            options[:default] = options[:default].to_yaml # ActiveRecord serialization is always stored as YAML
```

Comment:
I don't think this comment is fully accurate. If you pass a coder in (a class that responds to `load` and `dump`) then [Rails will use that as the serializer](https://github.com/rails/rails/blob/master/activerecord/lib/active_record/attribute_methods/serialization.rb#L41-L42) instead of their built in `YAMLColumn`. They even have a built in `JSON` coder that they use instead when you pass in the `json` gem as the coder for the column.

So for instance if I were to use the rails method like this:
```ruby
serialize :object_array, JSON
```
Rails would json serialize the column rather than yaml.  So I think we need to extract and parse the second argument to the serialize [similar to how Rails does it here](https://github.com/rails/rails/blob/master/activerecord/lib/active_record/attribute_methods/serialization.rb#L70-L76) in order to properly serialize the default.

Reply:
Hmm, I got that from the [docs](https://api.rubyonrails.org/classes/ActiveRecord/AttributeMethods/Serialization/ClassMethods.html):

> The serialization is done through YAML. If class_name is specified, the serialized object must be of that class on assignment and retrieval.

So when it's `Array` or `Hash`, they expose that type but store in YAML.

I assumed in your example of `JSON` that they would do the same thing. But you're right, they seem to store that in JSON.

I'll update the comment.

Reply:
I did a lot more than update the comment:
I added explicit test blocks for `JSON`, `Hash`, `Array`, and a custom `ValueClass` coder. And for each of those, I added tests for a variety of defaults, including the special `blank?` cases that get mapped to `NULL`. Please review the recent two commits.